recipes/qbs/all/conanfile.py

```diff
@@ -0,0 +1,125 @@
+        for dep, options in self._data["options"].items():
+            for key, value in options.items():
+                setattr(self.options[dep], key, value)
+
+        minimal_cpp_standard = "11"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+
+        minimal_version = {
+            "gcc": "5.0",
+            "clang": "6.0",
+            "apple-clang": "11",
+            "Visual Studio": "16",
+        }
+
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler standard version support".format(self.name, compiler))
+            self.output.warn(
+                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
+
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][0])
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch, base_path="qbs-src-%s" % self.version)
+
+    def build(self):
+        qmake = os.path.join(self.deps_cpp_info["qt"].bin_paths[0], "qmake")
+        source_dirpath = os.path.join(self.source_folder, "qbs-src-%s" % self.version)
+        project_filepath = os.path.join(source_dirpath, "qbs.pro")
+        args = [
+            "-r", project_filepath,
+            "QT-=gui",
+            "CONFIG+=release",
+            "CONFIG-=debug",
+            "CONFIG-=debug_and_release",
+            "CONFIG+=nomake_tests",
+            "CONFIG+=qbs_no_dev_install",
+            "CONFIG+=qbs_no_man_install",
+            "QBS_INSTALL_PREFIX=/"
+        ]
+        ncores = tools.cpu_count()
+
+        with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+            self.run("%s %s" % (qmake, " ".join(args)), run_environment=True)
+            self.run("%s -j%s" % (self._data["build_tool"], ncores), run_environment=True)
+
+    def package(self):
+        install_root = os.path.join(self.package_folder, "bin")
+        self.run("%s install INSTALL_ROOT=%s" % (self._data["build_tool"], install_root))
+        # This recipe builds Qbs statically, the libs are not needed.
+        tools.rmdir(os.path.join(install_root, "lib"))
+        tools.rmdir(os.path.join(install_root, "share", "qbs", "examples"))
+        self.copy(src="qbs-src-%s" % self.version, dst="licenses", pattern="LICENSE*")
+        self.copy(src="qbs-src-%s" % self.version, dst="licenses", pattern="LGPL*")
+
+    def package_id(self):
+        del self.info.settings.compiler
```

Comment:
@rweickelt can you also remove arch option or better use arch_build? Same for os option. You'll get troubles with cross-building profiles like android.

```
[settings]
arch=armv7
arch_build=x86_64
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=9
os=Android
os.api_level=21
os_build=Windows
[options]
[build_requires]
*: android_ndk_installer/r21b
[env]
```

Reply:
Have a look at this comment: https://github.com/conan-io/conan-center-index/pull/1716#discussion_r429558736

Reply:
Missed this comment, sorry